lib/compress.js

```diff
@@ -1042,6 +1049,18 @@ merge(Compressor.prototype, {
         def(AST_Constant, function(){
             return this.getValue();
         });
+        def(AST_Object, function(compressor){
```

Comment:
Probably need an `if (compressor.option("unsafe")) { ... }` wrapper around this logic similar to `def(AST_PropAccess` which calls `throw def;` if the option is not in effect.